### PR TITLE
Habitat real densities

### PR DESCRIPTION
**Pull request by PiezPiedPy**
Habitat now uses real densities and rates.
Inflatable habitats no longer consume nitrogen when disabled.
Inflatable habitats inflate 2x faster. (7 days for the Gravity Ring too inflate was silly)
Call Home is now disabled in the Planner when there is no external transmitter.
Plasma blackout now displays in Signal tooltip and messages.
Formatted the Supplies indicator tooltip.

### CODE REVIEW

File: src/UI/Planner.cs [hunk 1]
```diff
@@ -647,7 +647,7 @@ void Analyze_crew(List<Part> parts)
 		void Analyze_habitat(Resource_simulator sim, Environment_analyzer env)
 		{
 			// calculate total volume
-			volume = sim.Resource("Atmosphere").capacity;
+			volume = sim.Resource("Atmosphere").capacity / 1e3;
```
Comment (madman2003) [inline src/UI/Planner.cs]: 1e3 magic constant

File: GameData/Kerbalism/Profiles/Default.cfg [hunk 2]
```diff
@@ -147,7 +147,7 @@ Profile
             // The `input` parameter is used in conjunction with the `output_only` parameter to allow the rule to monitor the resource so that
             // it can display messages and allow the in-game panels to show the relevant information.
     monitor_offset = 0.6 // Base humidity of 60% - ***DO NOT CHANGE*** Bad things will happen as this value is Hard-Coded in Habitation.
-    rate = 0.000002879122904
+    rate = 0.002879122904
```
Comment (madman2003) [inline GameData/Kerbalism/Profiles/Default.cfg]: rate change but no comment change, what's going here?

File: GameData/Kerbalism/Profiles/Default.cfg [hunk 3]
```diff
@@ -214,19 +214,19 @@ Profile
...
+    output = CarbonDioxide@0.020201
     dump = true
   }
 
   // convention: 1 capacity = enough to compensate for leaks in 70 m^2 surface (a cube of 40 m^3 volume)
   Process
   {
     name = pressure control
-    modifier = _PressureControl
+    modifier = _PressureControl,breathable
```
Comment (madman2003) [inline GameData/Kerbalism/Profiles/Default.cfg]: what is the reason for the breathable modifier? (maybe add a comment?)

File: GameData/Kerbalism/Profiles/Default.cfg [hunk 4]
```diff
@@ -289,7 +288,7 @@ Profile
   {
     name = atmo leaks
     modifier = surface,breathable
-    input = Atmosphere@0.000000148
+    input = Atmosphere@0.000148
     // from ISS: 899 m³ volume, 452 m² surface (estimated), 0.227 Kg/day (structural) + 1.543 Kg/day (activities)
```
Comment (madman2003) [inline GameData/Kerbalism/Profiles/Default.cfg]: how is this input related to this comment? (considering 3600*6 seconds per day)

File: GameData/Kerbalism/System/Resources.cfg [hunk 5]
```diff
@@ -5,7 +5,7 @@
 RESOURCE_DEFINITION
 {
   name = Atmosphere
-  density = 0.001251          // 1 m^3 of nitrogen at STP, Density is a pseudo value for Habitation to work with percentages 
+  density = 0.000001251          // Nitrogen at STP
```
[the remark this answers is not in the document]
Comment (PiezPiedPy) [reply GameData/Kerbalism/System/Resources.cfg]: what do you mean ?

File: src/Lib.cs [hunk 6]
```diff
@@ -316,6 +316,14 @@ public static T PrivateField<T>(Type type, object instance, string field_name)
 			return (T)field.GetValue(instance);
 		}
 
+		// get access to a private field
+		public static T PrivateField<T>(object instance, string field_name)
```
[the remark this answers is not in the document]
Comment (PiezPiedPy) [reply src/Lib.cs]: I have no control over KSP's code, ask them to fix there bug

File: GameData/Kerbalism/System/Resources.cfg (hunk 5 above)
[the remark this answers is not in the document]
Comment (PiezPiedPy) [reply GameData/Kerbalism/System/Resources.cfg]: look at CRP

File: src/Modules/Habitat.cs [hunk 7]
```diff
@@ -126,14 +126,14 @@ public void Configure(bool enable)
 				{
 					// add internal atmosphere resources
 					// - disabled habitats start with zero atmosphere
-					Lib.AddResource(part, "Atmosphere", (state == State.enabled && Features.Pressure) ? volume : 0.0, volume);
-					Lib.AddResource(part, "WasteAtmosphere", 0.0, volume);
-					Lib.AddResource(part, "MoistAtmosphere", 0.0, volume); // 1 Kerbal = 0.05/day(5% @ 1m3) note. (base humidity is already at 0.6 = 60%)
+					Lib.AddResource(part, "Atmosphere", (state == State.enabled && Features.Pressure) ? volume * 1e3 : 0.0, volume * 1e3);
+					Lib.AddResource(part, "WasteAtmosphere", 0.0, volume * 1e3);
```
[the remark this answers is not in the document]
Comment (PiezPiedPy) [reply src/Modules/Habitat.cs]: volume is in m3 or 1000 Ltr, densities in KSP are for 1 Ltr

File: GameData/Kerbalism/Profiles/Default.cfg (hunk 3 above)
Comment (PiezPiedPy) [reply GameData/Kerbalism/Profiles/Default.cfg]: ask yourself what the breathable modifier does and you will have your answer. If you do not know then you will have to learn by reading the code, same as everyone else

Comment (PiezPiedPy) [reply GameData/Kerbalism/Profiles/Default.cfg]: I will make it easy for you this one time, in code the breathable modifier is true if the body a vessel is on has an atmosphere that is breathable, it is used by the rules and processes to determine if they should execute.
Having a breathable modifier attached to a rule or process in a .cfg file basically instructs that rule or process to stop running if it is in a breathable atmosphere.

File: GameData/Kerbalism/Profiles/Default.cfg (hunk 4 above)
Comment (steamp0rt) [reply GameData/Kerbalism/Profiles/Default.cfg]: oh wait he disappeared

Comment (PiezPiedPy) [reply GameData/Kerbalism/Profiles/Default.cfg]: lol

File: src/Modules/Habitat.cs [hunk 8]
```diff
@@ -198,7 +198,7 @@ State Equalize()
 
 				// determine equalization speed
 				// we deal with the case where a big hab is sucking all atmosphere from the rest of the vessel
-				double amount = Math.Min(partsHabVolume, volume) * equalize_speed * Kerbalism.elapsed_s;
+				double amount = Math.Min(partsHabVolume * 1e3, volume * 1e3) * (equalize_speed * 2) * Kerbalism.elapsed_s;
```
[the remark this answers is not in the document]
Comment (PiezPiedPy) [reply src/Modules/Habitat.cs]: it means multiply by 2, or equalize the habitat pressure twice as fast

File: src/UI/Planner.cs (hunk 1 above)
Comment (PiezPiedPy) [reply src/UI/Planner.cs]: magic is good, it makes things work
